grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -167,6 +167,12 @@ class LocationApiController extends BaseDomainApiController {
         if (!existingLocation) {
             throw new IllegalArgumentException("No Location found for location ID ${params.id}")
         }
+        // We only want to assign "None" if we go next from the Details step
+        // Without having this if, we would assign "None" when going from Address step to Details (previous step)
+        // because the jsonObject wouldn't contain supportedActivities
+        if (params.step == "1") {
```

Comment:
What is params.step? 

Aside: If you're basing this off of the use of step numbers for the stock movement wizard please know that I made a mistake there. We didn't really have a good understanding of how that should be done so that was admittedly kind of a hack. I would hate to repeat that mistake, so let's not do it here if there's a more explicit way of handling what we need. 

Feels like a good candidate for a tech huddle but in the meantime let's come up with something straightforward/explicit.

Reply:
Just so it's clear why this bothers me. Imagine someone developing a new client for OpenBoxes. A mobile application or god forbid, an Electron desktop application. They want to allow users to create new locations same as we do within the webapp. So they go to the API docs to see what they need to pass as a payload and they see an example.

> To create a location you'll want to post a request to the Locations API like this:

```
POST /openboxes/api/locations
{
        "name": "My New Location",
        "active": "true",
        "locationNumber": "MNL",
        "organization.id": null,
        "locationType.id": "2",
        "supportedActivities": null
}
```
> Oh and if you want to set the supported activities for this new location to empty, you need to pass a random parameter to the API like this:
```
POST /openboxes/api/locations?step=1
{
        "name": "My New Location",
        "active": "true",
        "locationNumber": "MNL",
        "organization.id": null,
        "locationType.id": "2",
        "supportedActivities": null
}
```
What I'm trying to say is we should be trying to think of our API endpoints from the perspective of other developers. Not what will work fine internally for us.

Reply:
I just walked through the Location Configuration wizard, set the required properties, unchecked use default activities, cleared the supported activities, and clicked Next.

![image](https://user-images.githubusercontent.com/1162935/231345808-f847bd6e-ba28-447e-b3d9-35efc62ea8bb.png)

And the logic for the step seemed to have been ignored. 

![image](https://user-images.githubusercontent.com/1162935/231346047-03d5d85e-02d5-41ac-a934-7d9d22c17978.png)

Here's the request as a cURL just so you can see what is being passed to the API by our react API client.

```
curl 'https://openboxes.ngrok.io/openboxes/api/locations?useDefaultActivities=false&step=1' \
  -H 'authority: openboxes.ngrok.io' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'content-type: application/json' \
  -H 'cookie: JSESSIONID=F3279E29D67D0DC79CC152D26033C3C1; __utmc=180537977; __utmz=180537977.1586446259.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none); JSESSIONID=2E25CB9439F5309BF0EF32D5B0F894B3; G_ENABLED_IDPS=google; uvts=c3af64d0-a29c-4dd6-5f32-6c516e52989b; __utma=180537977.890378389.1586446259.1679791756.1679802048.1740; _ga=GA1.3.890378389.1586446259; _gid=GA1.3.1054352685.1681247566; _gat_gtag_UA_37396941_1=1' \
  -H 'origin: https://openboxes.ngrok.io' \
  -H 'referer: https://openboxes.ngrok.io/openboxes/locationsConfiguration/create' \
  -H 'sec-ch-ua: "Chromium";v="112", "Google Chrome";v="112", "Not:A-Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Linux"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36' \
  --data-raw '{"name":"Test Location","active":true,"locationNumber":"TEST","organization.id":"5097","locationGroup.id":"","manager.id":"","locationType.id":"2","supportedActivities":[],"bgColor":"","fgColor":""}' \
  --compressed
```

Reply:
But regardless, what I think we should be doing is figuring out the most straightforward way to make the API do something explicit (i.e. override supported activities to be [None]) and require the UI and/or API client to figure out how to evaluate the current state, figure out whether conditions have been met and then pass that explicit requirement back to the API (make this ).

Note, that I say UI "or" API client. It might be that the API client can infer what we need from the current state without changing the UI. If not, then we need to consider changing the UI so the user is explicitly stating ... "i want this supported activities thing to be empty". It feels like we should be able to infer, but I haven't played around with the Location Configuration enough yet.

Reply:
> I just walked through the Location Configuration wizard, set the required properties, unchecked use default activities, cleared the supported activities, and clicked Next.

> And the logic for the step seemed to have been ignored.

That's weird and shouldn't have happened (should have None). Thanks for catching that and I'll double check that.

Reply:
According to the `params.step`, this is a first idea that came to my mind for handling this and also had in mind that it is used throughout the project. I didn't know you didn't like it and you would like to avoid that. Thanks for the explanation why it is wrong. I also have an idea how to handle that on the frontend side, but I'll probably have to write a bit more code.

To get you on the same page - there was a problem while going back to the details step from the address step, because when clicking "Previous" we are also saving the data, and because on the Address step we don't set the supportedActivities, it is not sent to the API while going back to the previous step, so the API overrode my supportedactivities to `["None"]` while doing so.

Reply:
@jmiranda but I can see that you've tested it on an environment where there are not my changes yet? Or am I mistaken?

Reply:
never mind, I haven't noticed the branch, I just saw: `https://openboxes.ngrok.io/openboxes/api/locations?useDefaultActivities=false&step=1` so I immedietaly assumed that there is not my branch deployed